crates/ruff_python_formatter/src/expression/expr_lambda.rs

```diff
@@ -62,7 +80,89 @@ impl FormatNodeRule<ExprLambda> for FormatExprLambda {
             }
         }
 
-        write!(f, [body.format()])
+        if is_parenthesize_lambda_bodies_enabled(f.context()) {
+            let fmt_body = format_with(|f: &mut PyFormatter| {
+                if matches!(&**body, Expr::Call(_) | Expr::Subscript(_)) {
+                    let unparenthesized = body.format().with_options(Parentheses::Never);
+                    if CallChainLayout::from_expression(
+                        body.into(),
+                        comments.ranges(),
+                        f.context().source(),
+                    ) == CallChainLayout::Fluent
+                    {
+                        parenthesize_if_expands(&unparenthesized).fmt(f)
+                    } else {
+                        let unparenthesized = unparenthesized.memoized();
+                        if unparenthesized.inspect(f)?.will_break() {
+                            expand_parent().fmt(f)?;
+                        }
+
+                        best_fitting![
+                            // body all flat
+                            unparenthesized,
+                            // body expanded
+                            group(&unparenthesized).should_expand(true),
+                            // parenthesized
+                            format_args![token("("), block_indent(&unparenthesized), token(")")]
+                        ]
+                        .fmt(f)
+                    }
+                } else if has_own_parentheses(body, f.context()).is_some()
```

Comment:
Let's add some documentation to those branches. What's the intended formatting for each of them, eg. For this brmach, the idea is to never add parantheses for expressions that start with a parentheses